adagrams/game.py

```diff
@@ -1,13 +1,120 @@
+    # builds 'letter_bank_count' to keep track of letters in the bank:
+    letter_bank_count = {}
+    for letter in letter_bank:
+        if letter not in letter_bank_count:
+            letter_bank_count[letter] = 1  # adds new key:value if not present
+        else:
+            letter_bank_count[letter] += 1  # adds to value if key is present
+    # builds 'word_letter_count' to keep track of letters in word:
+    word_letter_count = {}
+    for letter in formatted_word:
+        if letter not in word_letter_count:
+            word_letter_count[letter] = 1
+        else:
+            word_letter_count[letter] += 1
+        # checks if letter is present in letter bank OR if a letter has been
+        # used more than the number of times it appears in the bank; returns
+        # False if either condition is true:
+        if (letter not in letter_bank
+            or word_letter_count[letter] > letter_bank_count[letter]):
+            return False
+
+    return True  # f(x) returns True if checks above pass.
 
 def score_word(word):
-    pass
+    # dictionary of letters and their point values:
+    letter_values = {
+        'A': 1, 'B': 3, 'C': 3, 'D': 2, 'E': 1, 'F': 4, 'G': 2, 'H': 4, 'I': 1,
+        'J': 8, 'K': 5, 'L': 1, 'M': 3, 'N': 1, 'O': 1, 'P': 3, 'Q': 10, 'R':1,
+        'S': 1, 'T': 1, 'U': 1, 'V': 4, 'W': 4, 'X': 8, 'Y': 4, 'Z': 10
+    }
+    MIN_BONUS_LENGTH = 7  # minimum word length for length point bonus
+    WORD_LENGTH_BONUS = 8  # amount of points awarded for length bonus
+    score = 0
+
+    for letter in word:
+        formatted_letter = letter.upper()  # formats letter to uppercase
+
+        score += letter_values[formatted_letter]  # adds points to 'score'
+    # adds length bonus if word meets requirement:
+    if len(word) >= MIN_BONUS_LENGTH:
+        score += WORD_LENGTH_BONUS
+
+    return score
 
 def get_highest_word_score(word_list):
-    pass
\ No newline at end of file
+    # dictionary of letters and their point values:
+    letter_values = {
+        'A': 1, 'B': 3, 'C': 3, 'D': 2, 'E': 1, 'F': 4, 'G': 2, 'H': 4, 'I': 1,
+        'J': 8, 'K': 5, 'L': 1, 'M': 3, 'N': 1, 'O': 1, 'P': 3, 'Q': 10, 'R':1,
+        'S': 1, 'T': 1, 'U': 1, 'V': 4, 'W': 4, 'X': 8, 'Y': 4, 'Z': 10
+    }
+    MIN_BONUS_LENGTH = 7  # minimum word length for point bonus
+    WORD_LENGTH_BONUS = 8  # amount of points awarded for length bonus.
+    MAX_WORD_LENGTH = 10  # maximum word length permitted.
+    winning_word = ""  # tracks the winning word
+    highest_score = 0  # tacks the highest score
+
+    for word in word_list:
+        current_score = 0  # tracks the current score of a given word
+        # this for loop handles calculating the current word's score:
+        for letter in word:
+            current_score += letter_values[letter]
+        # adds the length bonus for 7 letter words or longer:
+        if len(word) > MIN_BONUS_LENGTH:
+            current_score += WORD_LENGTH_BONUS
```

Comment:
Just a heads up, your functions are allowed to build upon each other! It is perfectly ok for you to use the `score_word()` function that you created in wave 03 here. This avoids duplicated code.